dogen/plugins/artifact.py

```diff
@@ -0,0 +1,56 @@
+import glob
+import os
+import shutil
+
+from dogen.plugin import Plugin
+
+class Artifact(Plugin):
+    @staticmethod
+    def info():
+        return "artifact", "Support overriding the sources from an artifact directory"
+
+    @staticmethod
+    def inject_args(parser):
+        parser.add_argument('--artifact-local-enable', action='store_true', help='Uses local artifacts from the provided artifacts directory')
+        parser.add_argument('--artifact-local-dir', default='artifacts', help='Provides path to local artifacts directory')
+
+        return parser
+
+    def __init__(self, dogen, args):
+        super(Artifact, self).__init__(dogen, args)
+        self.artifact_files = []
+
+    def before_sources(self):
+
+        if not self.artifact_files:
+            return
+        
+        if not os.path.exists(self.output):
+            self.log.warn("build directory does not exist")
+            return
+        self.log.info("Copying local artifact files from '%s'" % self.args.artifact_local_dir)
+
+        for f in sorted(self.artifact_files):
+            self.log.debug("Copying local artifact %s" % os.path.basename(f))
+            shutil.copy2(f, self.output)
```

Comment:
The target name could be used in the artifacts directory. The sources would then find it and pick it up.  
I'll need to update to handle subdirectories.  If the artifacts directory contains the subdirectories and target names, they can be copied over.